src/test/java/com/google/devtools/build/lib/server/GrpcServerTest.java

```diff
@@ -460,6 +469,320 @@ public void onCompleted() {
         .isEqualTo(Code.INTERRUPTED);
   }
 
+  /**
+   * Ensure that if a command is marked as preemptible, running a second command
+   * interupts the first command
+   */
+  @Test
+  public void testPreeempt() throws Exception {
+    String firstCommandArg = "Foo";
+    String secondCommandArg = "Bar";
+
+    CommandDispatcher dispatcher =
+        new CommandDispatcher() {
+          @Override
+          public BlazeCommandResult exec(
+              InvocationPolicy invocationPolicy,
+              List<String> args,
+              OutErr outErr,
+              LockingMode lockingMode,
+              String clientDescription,
+              long firstContactTimeMillis,
+              Optional<List<Pair<String, String>>> startupOptionsTaggedWithBazelRc)
+              throws InterruptedException {
+            if (args.contains(firstCommandArg)) {
+              OutputStream out = outErr.getOutputStream();
+              try {
+                while (true) {
+                  if (Thread.interrupted()) {
```

Comment:
Here and others - can you just sleep and catch InterruptedException instead of busy waiting/writing? I don't think the `out.write`s are necessary for the test. Should simplify the code and give potentially under powered CI machines a break.

```
while (true) {
    try {
      Thread.sleep(TestUtils.WAIT_TIMEOUT_WHATEVER);
    } catch (InterruptedException e) {
       return BlazeCommandResult...
    }
}
```

I

Reply:
Definitely. I was using the logic from the interrupt stream tests, but it's not really necessary here.

Reply:
make the change here too?